classes/pdf/PDF.php

```diff
@@ -137,13 +137,6 @@ public function render($display = true)
                 continue;
             }
 
-            if (empty($this->filename)) {
-                $this->filename = $template->getFilename();
-                if (count($this->objects) > 1) {
-                    $this->filename = $template->getBulkFilename();
-                }
-            }
-
             $template->assignHookData($object);
```

Comment:
In previous implementation, when `assignHookData()` is called, `$pdf->filename` is set
In your implementation, when `assignHookData()` is called, `$pdf->filename` is not set

That could introduce issues for some modules.

Can I suggest we split the `$this->getFilename()` in two ? One `buildFilename()` and one `getFilename()`
```
    private function buildAndAssignFilename()
    {
        foreach ($this->objects as $object) {
            $template = $this->getTemplateObject($object);
            if (!$template) {
                continue;
            }

            $filename = $template->getFilename();
            if (count($this->objects) > 1) {
                $filename = $template->getBulkFilename();
            }
        }

        return $filename;
    }
```

Then we do
`if (empty($this->filename)) {$this->filename = $this->buildFilename()}` at line 140.

But we can still use `$this->getFilename()` in other lines
```
    public function getFilename()
    {
        if (empty($this->filename)) {
            $this->filename = $this->buildFilename();
        }
    

        return $this->filename;
    }
```

Reply:
I don't get why we would split it up. Why not just call `getFilename` just before assignHookData() ?
The method already returns the filename if it was already set.

Reply:
Now that I think of it, why not generate the filename at __construct? I'd say this should always be available right?

Reply:
> Now that I think of it, why not generate the filename at __construct? I'd say this should always be available right?
Indeed the sooner the better

Reply:
Fixed in https://github.com/PrestaShop/PrestaShop/pull/26587/commits/bd03107cfd421d36383f19001062fe47ba15cd93